apps/prairielearn/src/pages/instructorCourseAdminStaff/instructorCourseAdminStaff.html.ts

```diff
@@ -0,0 +1,748 @@
+                  )}
+                </select>
+                <select
+                  class="form-control form-control-sm"
+                  id="addUsersInputCourseInstanceRole"
+                  name="course_instance_role"
+                >
+                  <option value="Student Data Viewer" selected>Viewer</option>
+                  <option value="Student Data Editor">Editor</option>
+                </select>
+              </div>
+            </div>
+          `
+        : ''}
+
+      <div class="text-right mt-4">
+        <button type="button" class="btn btn-secondary" onclick="$('#${id}').popover('hide')">
+          Cancel
+        </button>
+        <button type="submit" class="btn btn-primary">Add users</button>
+      </div>
+    </form>
+  `;
+}
+
+function CoursePermissionsDeleteNonOwnersForm({
+  csrfToken,
+  id,
+}: {
+  csrfToken: string;
+  id: string;
+}) {
+  return html`
+    <form name="delete-non-owners" method="POST">
+      <input type="hidden" name="__action" value="delete_non_owners" />
+      <input type="hidden" name="__csrf_token" value="${csrfToken}" />
+
+      <div class="form-group mb-4">
+        <p class="form-text">
+          Taking this action will remove every user from course staff who is not a course content
+          Owner.
+        </p>
+      </div>
+
+      <div class="text-right mt-4">
+        <button type="button" class="btn btn-secondary" onclick="$('#${id}').popover('hide')">
+          Cancel
+        </button>
+        <button type="submit" class="btn btn-primary">Delete non-owners</button>
+      </div>
+    </form>
+  `;
+}
+
+function StaffTable({
+  csrfToken,
+  courseUsers,
+  authnUser,
+  user,
+  isAdministrator,
+}: {
+  csrfToken: string;
+  courseUsers: CourseUsersRow[];
+  authnUser: User;
+  user: User;
+  isAdministrator: boolean;
+}) {
+  return html`
+    <table class="table table-sm table-hover table-striped">
+      <thead>
+        <th>UID</th>
+        <th>Name</th>
+        <th>Course content access</th>
+        <th>Student data access</th>
+        <th>Actions</th>
+      </thead>
+      <tbody>
+        ${courseUsers.map((courseUser) => {
+          if (!courseUser.user.name) hasUnknownUsers = true;
+          return html`
+            <tr>
+              <td class="align-middle">${courseUser.user.uid}</td>
+              <td class="align-middle">
+                ${courseUser.user.name
+                  ? courseUser.user.name
+                  : html`<span class="text-danger">Unknown user</span>`}
+              </td>
+              <td class="align-middle">
+                ${(courseUser.user.user_id === authnUser.user_id ||
+                  courseUser.user.user_id === user.user_id) &&
+                !isAdministrator
+                  ? //     Cannot change the course role of
+                    // yourself (or of the user you are // emulating) unless you are an administrator.
```

Comment:
This gets kind of tricky to read, consider lifting up the conditional into a variable. Please double check that I inverted the logic correctly:

```suggestion
          if (!courseUser.user.name) hasUnknownUsers = true;
          // Cannot change the course role of yourself (or of the
          // user you are emulating) unless you are an administrator.
          const canChangeCourseRole = (courseUser.user.user_id !== authnUser.user_id &&
                  courseUser.user.user_id !== user.user_id) ||
                isAdministrator;

          return html`
            <tr>
              <td class="align-middle">${courseUser.user.uid}</td>
              <td class="align-middle">
                ${courseUser.user.name
                  ? courseUser.user.name
                  : html`<span class="text-danger">Unknown user</span>`}
              </td>
              <td class="align-middle">
                ${!canChangeCourseRore
                  ?
```

This will of course need some Prettier action!

Reply:
Updated to this in [38466f5](https://github.com/PrairieLearn/PrairieLearn/pull/9822/commits/38466f5b501777db47a1f16f8668ec8e2162777a).